src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -297,86 +297,139 @@ def test_get_lbound_expression():
+    dtsym = DataSymbol("oops", DeferredType())
+    dtref = ArrayReference.create(dtsym,
+                                  [_ONE.copy(), _ONE.copy(), _ONE.copy()])
+    ubnd = dtref._get_bound_expression(1, "upper")
+    assert isinstance(ubnd, BinaryOperation)
+    assert ubnd.operator == BinaryOperation.Operator.UBOUND
+    assert ubnd.children[0].symbol is dtsym
+    assert ubnd.children[1] == Literal("2", INTEGER_TYPE)
+
 
 @pytest.mark.parametrize("extent", [ArrayType.Extent.DEFERRED,
                                     ArrayType.Extent.ATTRIBUTE])
-def test_get_lbound_expression_unknown_size(extent):
+def test_get_bound_expression_unknown_size(extent):
     '''
-    Test the get_lbound_expression method when we have the definition of the
+    Test the _get_bound_expression method when we have the definition of the
     array type but its dimensions are unknown.
 
     '''
     symbol = DataSymbol("my_symbol", ArrayType(INTEGER_TYPE,
                                                [extent, extent]))
     aref = ArrayReference.create(symbol, [_ONE.copy(), _ONE.copy()])
-    lbnd = aref.get_lbound_expression(1)
+    lbnd = aref._get_bound_expression(1, "lower")
     assert isinstance(lbnd, BinaryOperation)
     assert lbnd.operator == BinaryOperation.Operator.LBOUND
     assert lbnd.children[0].symbol is symbol
 
+    symbol = DataSymbol("my_symbol", ArrayType(INTEGER_TYPE,
+                                               [extent, extent]))
+    aref = ArrayReference.create(symbol, [_ONE.copy(), _ONE.copy()])
+    ubnd = aref._get_bound_expression(1, "upper")
+    assert isinstance(ubnd, BinaryOperation)
+    assert ubnd.operator == BinaryOperation.Operator.UBOUND
+    assert ubnd.children[0].symbol is symbol
 
-def test_aref_to_aos_lbound_expression():
+
+def test_aref_to_aos_bound_expression():
     '''
-    Test the get_lbound_expression() method for an ArrayReference to an array
+    Test the _get_bound_expression() method for an ArrayReference to an array
     of structures.
 
     '''
     sgrid_type = StructureType.create(
-        [("id", INTEGER_TYPE, Symbol.Visibility.PUBLIC)])
+        [("ID", INTEGER_TYPE, Symbol.Visibility.PUBLIC)])
     sgrid_type_sym = DataTypeSymbol("subgrid_type", sgrid_type)
     sym = DataSymbol("subgrids", ArrayType(sgrid_type_sym, [(3, 10)]))
-    one = Literal("1", INTEGER_TYPE)
     lbound = BinaryOperation.create(BinaryOperation.Operator.LBOUND,
-                                    Reference(sym), one)
+                                    Reference(sym), _ONE.copy())
     ubound = BinaryOperation.create(BinaryOperation.Operator.UBOUND,
-                                    Reference(sym), one.copy())
+                                    Reference(sym), _ONE.copy())
     array = ArrayReference.create(sym, [Range.create(lbound, ubound)])
-    lbnd = array.get_lbound_expression(0)
+    lbnd = array._get_bound_expression(0, "lower")
     assert lbnd.value == "3"
 
+    sgrid_type = StructureType.create(
```

Comment:
Unless I'm missing something, aren't L393-L401 duplicates of L381-L389?

Reply:
Yes, removed.